core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -595,8 +595,10 @@ private[log] class Cleaner(val id: Int,
       log.replaceSegments(List(cleaned), segments)
     } catch {
       case e: LogCleaningAbortedException =>
-        try cleaned.deleteIfExists()
-        catch {
+        try {
+          cleaned.deleteIfExists()
+          log.producerStateManager.deleteIfExists(cleaned.baseOffset)
```

Comment:
Hmm, the cleaned segment has the same base offset as the first segment. So, we don't want to delete that snapshot file.